draft-ietf-quic-multipath.md

```diff
@@ -314,6 +314,18 @@ Each endpoint associates a Receiver Packet Number space to each Path Identifier
 that it provides to the peer. Each endpoint associates a Sender Packet Number space 
 to each Path Identifier received from the peer.
 
+Endpoints use the same Path ID for one specific path in both directions. 
+For a client-initiated path, the client decides which Path ID used for the new path, 
+it picks one of the server allocated CID with the specified Path ID. 
+Then the client send a PATH_CHALLENGE with the chosen CID. If the server receives the PATH_CHALLENGE, 
+it picks a Connection ID with the same path ID for sending the PATH_RESPONSE.
+
+The client MUST choose a previously unused Path ID for which both endpoints have already issued at least one connection ID.
+If the server receives a PATH_CHALLENGE before receiving MP_NEW_CONNECTION_ID 
+for the specific path, it MAY choose to ignore the PATH_CHALLENGE, or it can
+choose to send the PATH_RESPONSE frame upon reception of a
+MP_NEW_CONNECTION_ID frame containing the corresponding Path ID.
```

Comment:
I think the client should only try to open a path if it had send CIDs for that path ID (and the reception has been ack'ed). Then if the server received a path challenges for a path ID it doesn't have a CID for, it should either be an error or it should send a path abandon on another path.

Reply:
> I think the client should only try to open a path if it had send CIDs for that path ID (and the reception has been ack'ed).

We could be in the situation where the server has sent MP_NEW_CONNECTION_ID frames to the client, and the client wants to open a new path by sending a PATH_CHALLENGE along with a MP_NEW_CONNECTION_ID frame with the associated Path ID.

Reply:
Yes sending it in the same packet is fine as well because than it cannot be lost or reordered separately.

Reply:
Actually this issue is not resolved because if you send it in the same packet, it's "just" a matter of processing it the right order. However, otherwise I don't think it's a good idea to save the path challenge for later because then you also have to specify how long to wait.

Reply:
I removed this text for now. I will create a separate PR (and issue) for this as soon as we have merged this PR.